File: typescript/resistor-color/resistor-color.ts
export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];

export function colorCode(color: typeof COLORS[number]): number {
  return COLORS.indexOf(color);
}

File: kotlin/resistor-color/src/main/kotlin/ResistorColor.kt
object ResistorColor {
    enum class Colors {
        black,
        brown,
        red,
        orange,
        yellow,
        green,
        blue,
        violet,
        grey,
        white
    }

    fun colorCode(input: String): Int = Colors.valueOf(input).ordinal

    fun colors(): List<String> = Colors.values().map { color -> color.name }
}
